MiniSim/Menu.swift

```diff
@@ -162,11 +153,8 @@ class Menu: NSMenu {
             menuItem.submenu = isAndroid ? populateAndroidSubMenu(booted: device.booted) : populateIOSSubMenu(booted: device.booted)
             menuItem.state = device.booted ? .on : .off
             
-            DispatchQueue.main.async {
-                let iosDevicesCount = self.devices.filter({ $0.platform == .ios }).count
-                self.safeInsertItem(menuItem, at: isAndroid && UserDefaults.standard.enableiOSSimulators ? (isFirst ? index : iosDevicesCount) + 3 : 1)
-            }
-            
+            let iosDevicesCount = self.devices.filter({ $0.platform == .ios }).count
```

Comment:
Also here we are missing the jump back to the main thread. You could add the `DispatchQueue.main.async` to the `safeInsertItem` method.